ONSdigital/dp-legacy-cache-api
# Feature/api write authentication without permissions

### What

Added authentication to endpoints

### How to review

- run make test - should not fail
- then test manually with postman or curl - I have a token created here that is valid I can direct message you in slack
    -  have mongo db running in docker container
    -  run zebedee run script ./run.sh - not the zebedee reader - because middleware is still checking for each request if it contains a token
    -  run make debug
    -  get request to : http://localhost:29100/v1/cache-times/a1b2c3d4e5f67890123456789abcdef0 - should return 200 with or without token
    -  put request to : http://localhost:29100/v1/cache-times/a1b2c3d4e5f67890123456789abcdef0 - should return 401 without token - if you have a token put this in the header
    curl -i http://username:password@dev.myapp.com/api/users -H "Authorization: Bearer <token here>" -d 
{
    "path": "newcachetimepath",
    "etag": "newcachetimeetag",
    "collection_id": 1234,
    "release_time": "2023-12-12T00:00:00.000Z"
}"



### Who can review

Describe who worked on the changes, so that other people can review.
